cli/internal/taskhash/taskhash.go

```diff
@@ -62,44 +62,13 @@ func NewTracker(rootNode string, globalHash string, pipeline fs.Pipeline) *Track
 	}
 }
 
-// packageFileSpec defines a combination of a package and optional set of input globs
-type packageFileSpec struct {
-	pkg    string
-	inputs []string
+// packageFileHashInputs defines a combination of a package and optional set of input globs
+type packageFileHashInputs struct {
+	taskID         string
+	taskDefinition *fs.TaskDefinition
+	packageName    string
 }
 
-func specFromPackageTask(packageTask *nodes.PackageTask) packageFileSpec {
-	return packageFileSpec{
-		pkg:    packageTask.PackageName,
-		inputs: packageTask.TaskDefinition.Inputs,
-	}
-}
-
-// packageFileHashKey is a hashable representation of a packageFileSpec.
-type packageFileHashKey string
-
-// hashes the inputs for a packageTask
-func (pfs packageFileSpec) ToKey() packageFileHashKey {
-	sort.Strings(pfs.inputs)
-	return packageFileHashKey(fmt.Sprintf("%v#%v", pfs.pkg, strings.Join(pfs.inputs, "!")))
```

Comment:
This _probably_ can't go wrong, but we already have a unique identifier, the taskID. This was to enable multiple tasks in the same package with the same specified inputs to not have to rerun file hashing.

It's an optimization, but not actually the ideal one. Now that everything calls into `GetFileHashes` that could just maintain a set of all files it has seen and apply it across all packages, all inputs.

I do not add this feature in this PR.